PencilCase/PencilCase/scripts/Navigator.ts

```diff
@@ -0,0 +1,104 @@
+﻿import { ProductManagement } from './Pages/ProductManagement';
+import { PageBase } from './Pages/PageBase';
+import { HomePage } from './Pages/HomePage';
+import { Retail } from './Pages/Retail';
+import { Application } from './application';
+import * as Consts from './Pages/Consts';
+
+export class Navigator {
+    // TODO: Pop up pages/dialogs handle 
+
+    private static _instance: Navigator;
+    private tempPage: PageBase;
+
+    public static get instance(): Navigator {
+        if (Navigator._instance == null) {
+            Navigator._instance = new Navigator();
+        }
+
+        return Navigator._instance;
+    }
+
+    //public navigateTo = (toPage: string | JQuery, options?: any) => {
+    // In JQuery Mobiel doc, the toPage can be string or JQuery object, but for my case,
+    // the navigation only works with JQuery object, it must be due to the navigation ways I implemented.
+    // So I restraint the type to be JQuery object here to avoid spending time on debugging.
+    // Whatever, I don't want to spend more time on it since it already took me much time to make everything works as it does currently
+    public navigateTo = (toPage: JQuery, options?: any) => {
+        (<any>($)).mobile.changePage(toPage, options);
+    }
+
+    public goHome = () => {
+        Application.instance.activePage(new HomePage());
+        $(':mobile-pagecontainer').pagecontainer("change", "#HomePage");
+    }
+
+    public initialize = () => {
+        $(':mobile-pagecontainer').on("pagecontainerbeforechange", (eventObject: JQueryEventObject, parameters: any) => {
+            if (parameters.toPage !== Consts.Pages.HomePage.Id) { // No need the handling here for home page
```

Comment:
Home page is handled as a field of Application, no need to handle it here.
And tried to handle it in the same way as other pages caused navigation problems, due to you couldn't set options when change page, when trying to navigate back to home page from other pages like retail page, that will load another identical home page in the <body> except for different data-url, and has KO binding issue. Caos